### ISSUE TITLE
Upstream alignment

### ISSUE THREAD

**Issue by metux** [marchaesen/vcxsrv#32]
Hello,

I'm writing you as a member of Xorg upstream and I'd like to get in touch for coordination between you and us.
There's a lot happening in master branch might be relevant here.

One particular aspect to discuss: we're having MRs in the pipeline for removing remains of Cygwin support (at least lots of ifdef's for it). Also seen you've changed some pieces in in xwin - maybe those could be upstreamed ?

 Let's talk about that on xorg-devel mailing list :)

Also having an inofficial chat group on telegram: https://t.me/x11dev

have fun,
--mtx

**Comment by metux** [comment]
my mail addr: info@metux.net

**Comment by sskras** [comment]
Sounds intriguing. :)

Could anyone compare VcXsrv feature-wise with Cygwin's [XWin](https://gitlab.freedesktop.org/xorg/xserver/-/commits/master/hw/xwin?ref_type=heads) which live in the official repo of Xorg, please?

**Comment by metux** [comment]
> Sounds intriguing. :)
> 
> Could anyone compare VcXsrv feature-wise with Cygwin's [XWin](https://gitlab.freedesktop.org/xorg/xserver/-/commits/master/hw/xwin?ref_type=heads) which live in the official repo of Xorg, please?

It's actually not cygwin, but mingw32.

**Comment by sskras** [comment]
@metux, oh. 

Does this mean that Cygwin's page is wrong? Quoting [Development - XWin Server](https://x.cygwin.com/devel/server/#:~:text=View%20the%20XWin%20Server%20code%20in%20the%20X.org%20git%20tree%20under%20hw/xwin/):

> ### Getting Started
> * [View the XWin Server code](http://cgit.freedesktop.org/xorg/xserver/tree/) in the X.org git tree under [hw/xwin/](http://cgit.freedesktop.org/xorg/xserver/tree/hw/xwin)
> * Install Cygwin and Cygwin/X using Cygwin's setup. The [Cygwin/X User's Guide](https://x.cygwin.com/docs/ug/) may be helpful.

**Comment by efa** [comment]
I think the request was to do a comparison between the features of the X server provided with Cygwin (called XWin) and vcxsrv.

Not to compile the vcxsrv server with Cygwin

**Comment by metux** [comment]
> I think the request was to do a comparison between the features of the X server provided with Cygwin (called XWin) and vcxsrv.
> 
> Not to compile the vcxsrv server with Cygwin

Correct. The idea is aligning both sides so vcxsrv can directly build from Xorg upstream with minimal extra patches.
(or hopefully, some day both becoming one ;-))
